prog/page_nexus.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class Prog::PageNexus < Prog::Base
+  semaphore :resolve
+
+  def self.assemble(summary)
+    DB.transaction do
+      p = Page.create(summary: summary)
+
+      Strand.create(prog: "PageNexus", label: "start") { _1.id = p.id }
+    end
+  end
+
+  def page
+    @page ||= Page[strand.id]
+  end
```

Comment:
Progs have `subject_is` helper. Since strand and page have same id you can use `subject_is :page` instead of defined this method.